libraries/chain/services/betting_service.cpp

```diff
@@ -18,5 +68,115 @@ bool dbs_betting::is_betting_moderator(const account_name_type& account_name) co
     }
     FC_CAPTURE_LOG_AND_RETHROW((account_name))
 }
+
+const bet_object& dbs_betting::create_bet(const account_name_type& better,
+                                          const game_id_type game,
+                                          const wincase_type& wincase,
+                                          const std::string& odds_value,
+                                          const asset& stake)
+{
+    try
+    {
+        FC_ASSERT(stake.amount > 0);
+        FC_ASSERT(stake.symbol() == SCORUM_SYMBOL);
+        return _bet.create([&](bet_object& obj) {
+            obj.created = _dgp_property.head_block_time();
+            obj.better = better;
+            obj.game = game;
+            obj.wincase = wincase;
+            obj.value = odds::from_string(odds_value);
+            obj.stake = stake;
+            obj.rest_stake = stake;
+            obj.potential_gain = calculate_gain(obj.stake, obj.value);
+        });
+    }
+    FC_CAPTURE_LOG_AND_RETHROW((better)(game)(wincase)(odds_value)(stake))
+}
+
+void dbs_betting::match(const bet_object& bet)
+{
+    try
+    {
+        FC_ASSERT(is_need_matching(bet));
+
+        using pending_bets_type = std::vector<pending_bet_service_i::object_cref_type>;
+
+        pending_bets_type removing_pending_bets;
+
+        _pending_bet.foreach_pending_bets(bet.game, [&](const pending_bet_object& pending) -> bool {
+
+            const bet_object& pending_bet = _bet.get_bet(pending.bet);
+
+            if (is_bets_matched(bet, pending_bet))
+            {
+                auto matched_stake
+                    = calculate_matched_stake(bet.rest_stake, pending_bet.rest_stake, bet.value, pending_bet.value);
+                if (matched_stake.amount > 0)
+                {
+                    _bet.update(pending_bet, [&](bet_object& obj) {
+                        obj.rest_stake.amount = 0;
+                        obj.gain += matched_stake.amount;
+                    });
+                    _bet.update(bet, [&](bet_object& obj) {
+                        obj.rest_stake.amount -= matched_stake.amount;
+                        obj.gain += pending_bet.stake;
+                    });
+                }
+                else if (matched_stake.amount < 0)
+                {
+                    _bet.update(bet, [&](bet_object& obj) {
```

Comment:
`_bet.update` looks like you're trying to update a `_bet`. It is better to name a service instance like a `_bet_service`. Same for the rest services.